bin/utest/signal.c

```diff
@@ -199,9 +199,12 @@ int test_signal_mask(void) {
   sigcont_handled = 0;
   kill(pid, SIGUSR1);
 
-  /* Yield a couple times to make sure that the child sends us the signal. */
-  for (int i = 0; i < 3; i++)
+  /* Wait until we get a signal from the child. */
+  sigset_t set;
+  do {
     sched_yield();
```

Comment:
Could we use `sigsuspend` here instead?

Reply:
We could, but then the handler would run immediately after `sigsuspend()`, and in this test I want to check that the handler runs immediately after unblocking the signal with `sigprocmask()`.

Reply:
Right, I see.